File: example/structs/struct.js
var FFI = require("../../lib/ffi");

var Test = FFI.Struct([
    ['int32', 'a']
  , ['double', 'b']
  , ['string', 'c']
]);

var libstruct = new FFI.Library("./libstruct", {
    test_struct_arg_by_value: [ 'double', [ Test ] ]
  , test_struct_rtn_by_value: [ Test, [ ] ]
});

var output = libstruct.test_struct_rtn_by_value()

console.log(output.a)
console.log(output.b)
console.log(output.c)

libstruct.test_struct_arg_by_value(output)

File: example/structs/struct.c
#include <stdio.h>

struct test {
  int a;
  double b;
  char *c;
};


double test_struct_arg_by_value (struct test by_value) {
  printf("Input c: %s\n", by_value.c);
  return by_value.a + (double)by_value.b;
}

struct test test_struct_rtn_by_value () {
  struct test rtn;
  rtn.a = 10;
  rtn.b = 123.213;
  rtn.c = "Does this work?";
  return rtn;
}
